api/src/services/authentication.ts

```diff
@@ -74,6 +74,7 @@ export class AuthenticationService {
 			accountability: this.accountability,
 			status: 'pending',
 			user: user?.id,
+			obj: user,
```

Comment:
Instead of a new `obj` property, lets just not omit stuff from the `hookPayload` 👍🏻

Reply:
The intent of use `obj` and pass `user` is to modify directly the `user.password` property so it will not fail here:
https://github.com/directus/directus/blob/13db4c77193970614c20be09eb6f36726296f8f1/api/src/services/authentication.ts#L128

Reply:
Oh we shouldn't do that mutable! Instead, lets catch the altered user payload coming from the emitter and use that to login with